text/0000-ubi-based-stacks.md

```diff
@@ -0,0 +1,181 @@
+# Stacks & Extensions for UBI base images. (UBI8)
+
+## Summary
+
+A set of stacks based on the UBI base image should be developed, released, and maintained. 
+
+Unlike the existing stacks, that come in "full", "base", and "tiny" the UBI stack will initially start with a 
+single variant, with future variants to follow. The initial variant will have buildpacks for a handpicked set of 
+runtimes that have been tested with UBI, and will make use of a number of CNCF Buildpack 'Extensions' that will be 
+developed, released and maintained by the relevant teams to offer installation of selected runtimes and dependencies 
+via rpm/dnf. These extensions will be packaged with the UBI builder to allow the customer the option of using the 
+UBI provided runtimes. 
+
+## Motivation
+
+Red Hat Universal Base Images (UBI) are designed to be a foundation for cloud-native and web applications use cases 
+developed in containers. UBI Images form a popular choice for base container images, and can even be preferred 
+within environments with existing UBI deployments. In addition, use of a UBI image can sometimes be a requirement 
+for an application qualifying for support from some vendors while in production. That requirement is also extended 
+to any runtime in use, eg for Node or Java, the runtime must be installed via rpm/dnf, rather than via a tarball. 
+
+## Detailed Explanation
+
+### Background
+
+There are a wide variety of UBI images offered by Red Hat, offering pre configured deployments of various runtimes 
+(such as Java), and offering cutdown images designed for extension (ubi-minimal).
+
+While it would be possible to use UBI images in a similar fashion to the existing Ubuntu based images, and have 
+buildpacks perform the download & install of depdencies within layers, the preferred route would be to to use the 
+dependencies supplied via the configured package repositories within the images. Eg, rather than downloading a 
+JVM/JDK, one would be installed via 'yum' or 'dnf'. 
```

Comment:
I am not sure that I fully understand why installation from a configured package this is preferable method? Could you layout what it is superior to using the CNB layer system that we are currently using?

Reply:
I'm not sure it's that simple, to be able to say one is superior to the other.. The tarball+layers dir approach is perfectly capable of creating a working functional application, and so is the route where the runtime is installed via system packages.

I explain the motivation behind wishing to use system based deps within the section "UBI without rpms" a little further on in the RFC, but in essence, there's a support driven motivation here, that in order to be totally supported, a customer on UBI needs to also be using a UBI installed runtime (eg, java, or node, which are the ones being considered here). 

As we consider this to be one of the drivers for for UBI based apps, that's why having these runtimes installed in this manner becomes the default for the UBI stack. That's not to say it can't be overridden, but out of the box, this builder aims to create images that are fully eligible for support. (Or at least as far as they can be within the context of the build & run components managed by Paketo).

Reply:
> I explain the motivation behind wishing to use system based deps within the section "UBI without rpms" a little further on in the RFC, but in essence, there's a support driven motivation here, that in order to be totally supported, a customer on UBI needs to also be using a UBI installed runtime (eg, java, or node, which are the ones being considered here).

Gotcha! I just went back and looked over that section again and I think that makes sense. I guess I would be curious to see how this work and the understand more the drawbacks of using `Extensions` because I think that having two different types of buildpack like resource for installing dependencies could get confusing and I would love to potentially consolidate on one or the other. I don't think however answering that question should block this RFC so I think that everything else from my perspective as a builder maintainer makes sense!